hazelcast/src/main/java/com/hazelcast/jet/impl/connector/HazelcastReaders.java

```diff
@@ -336,19 +327,39 @@ public int getClassId() {
         }
     }
 
+    /**
+     * Create a ProcessorSupplier for a remote map in another cluster
+     */
     @Nonnull
-    public static <K, V, T> ProcessorSupplier readRemoteMapSupplier(
-            @Nonnull String mapName,
-            @Nonnull ClientConfig clientConfig,
-            @Nonnull Predicate<? super K, ? super V> predicate,
-            @Nonnull Projection<? super Entry<K, V>, ? extends T> projection
-    ) {
-        checkSerializable(Objects.requireNonNull(predicate), "predicate");
-        checkSerializable(Objects.requireNonNull(projection), "projection");
-
-        String clientXml = ImdgUtil.asXmlString(clientConfig);
-        return new RemoteProcessorSupplier<>(clientXml, new RemoteMapQueryReaderFunction<>(mapName, predicate,
-                projection));
+    public static <T, K, V> ProcessorSupplier readRemoteMapSupplier(RemoteMapSourceParams<T, K, V> params) {
+        if (params.hasDataSourceConnection()) {
+            if (params.hasPredicate()) {
+                checkSerializable(params.getPredicate(), "predicate");
+                checkSerializable(params.getProjection(), "projection");
+
+                RemoteMapQueryReaderFunction<K, V, T> readerSupplier = new RemoteMapQueryReaderFunction<>(
```

Comment:
I'd say this is a perfect example where using `var` simplifies the code

Reply:
rev : f5ce6365c3b2e9ae2117fb283939eb29e1aa2224